src/WebJobs.Script/Description/DotNet/Compilation/CSharp/Analyzers/InvalidPrivateMetadataReferenceAnalyzer.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.Azure.WebJobs.Script.Description.DotNet.CSharp.Analyzers
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    public class InvalidPrivateMetadataReferenceAnalyzer : DiagnosticAnalyzer
+    {
+        private const string Title = "Invalid private metadata reference";
+        private const string MessageFormat = "The reference '{0}' is invalid. Private assembly references must include the file extension. Consider using '{1}'.";
```

Comment:
nit: "Consider" makes it sound like this is optional or something. How about "Try using"

Reply:
Done